File: An2/MAP/MAP-Java/MAP-Lab7/src/Model/UndergraduateStudent.java
package Model;

/**
 * Created with IntelliJ IDEA.
 * User: mihai
 * Date: 11/28/13
 * Time: 3:39 PM
 * To change this template use File | Settings | File Templates.
 */
public class UndergraduateStudent extends Student {
    private int grade2;

    public UndergraduateStudent(int id, String name, int grade, int grade2){
        super(id, name, grade);
        this.grade2 = grade2;
    }

    @Override
    public float average(){
        return (this.getGrade() + this.grade2) / 2;
    }

    @Override
    public boolean isGreaterThan(Student student){
        return (this.average() > student.average());
    }

    @Override
    public String toString(){
        return this.getID() + " " + getName() + " " + this.getGrade() + " " + grade2;
    }
}

File: An2/Sem1/MAP/MAP-C#/MAP-Lab8/MAP-Lab8/Model/UndergraduateStudent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_Lab8.Model
{
    [Serializable()]
    class UndergraduateStudent : Student
    {
        public int grade2;

        public UndergraduateStudent(int id, String name, int grade, int grade2)
            : base(id, name, grade)
        {
            this.grade2 = grade2;
        }

        public override float average()
        {
            return (this.getGrade() + this.grade2) / 2;
        }


        public override String ToString()
        {
            return this.getID() + " " + this.getName() + " " + this.getGrade() + " " + this.grade2;
        }
    }
}
